tests/test_api/test_endpoints.py

```diff
@@ -173,6 +174,71 @@ def get_token():
         )
 
 
+def test_update_user():
+    mdb = get_mongo(run_config_frozen__normal_env).db
+    rs = ensure_test_resources(mdb)
+    base_url = os.getenv("API_HOST")
+
+    @retry(wait=wait_random_exponential(multiplier=1, max=60))
```

Comment:
FYI: The [docs](https://pypi.org/project/retry/) of the `retry` decorator say:

> ```
> :param tries: the maximum number of attempts. default: -1 (infinite).
> ```

I wonder whether this can lead to tests hanging indefinitely. I think that is what @shreddd was alluding to [here](https://github.com/microbiomedata/nmdc-runtime/issues/778#issuecomment-2480247793).

Reply:
I was wondering if we need the retry in the first place, but in talking to @dwinston this may be needed to account for async code. The tests shouldn't hang infinitely because it has a 60 second timeout.

Reply:
I think the timeout (which can be up to 60 seconds long) can happen an infinite number of times (because the default value of `tries` is `-1`). That is where I got the idea of hanging indefinitely from.

Reply:
In other words, the duration of each thing is limited; but the quantity of things is currently unlimited.

Reply:
Ah I see. We can add `stop=stop_after_attempt(N))` as an argument to retry. What is a good value for this N? Maybe 3 or 5?

Reply:
OK - added stop=stop_after_attempt(3)

Reply:
Thanks, @shreddd. Stopping after three attempts sounds good to me.

Turns out I was looking at the documentation for the wrong PyPI package. Now that I've looked at the `import` statements in this Python file (I'm sorry I didn't do that earlier), I see that the package being used is `tenacity` (its documentation is [here](https://tenacity.readthedocs.io/en/stable/api.html)). I had incorrectly assumed that it was the older library called `retry` (which is documented [here](https://github.com/invl/retry)). That's why I kept referring to a kwarg named `tries` (instead of `stop`). By coincidence, they both do default to an infinite number of attempts (whether framed as "try infinity times" or "never stop trying").